benchmarks/nemo/nemolite2d/manual_versions/psykal_c_opencl/time_step.c

```diff
@@ -784,11 +786,20 @@ void c_invoke_time_step(
 // Function that specify how to retrieve the device data 'from' to a host
 // location 'to', this function will be called by the infrastructure whenever
 // the data is needed on the host.
-void c_read_from_device(cl_mem from, double * to, int nx, int ny, int width){
-    cl_event read_events[1];
-    // Use width instead of nx in case there are padding elements
-    int ret = clEnqueueReadBuffer(command_queue[0], from, CL_TRUE, 0,
-			    width*ny, to, 0, NULL, &(read_events[0]));
-    check_status("clEnqueueReadBuffer", ret);
-    clWaitForEvents(1, read_events);
+void c_read_from_device(void * from, void * to, int startx, int starty, int nx,
+                        int ny, bool blocking){
+    int size_in_bytes = buff_size;
```

Comment:
Tbh, this can be computed from nx and ny, so I removed the global variable now.

However I added a comment in both saying that this implementation is only for blocking full buffer copies, as these are the only ones happening on the manual implementation.